js-api-spec/deprecations.test.ts

```diff
@@ -0,0 +1,308 @@
+// Copyright 2024 Google LLC. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import {
+  compileString,
+  deprecations,
+  Deprecation,
+  Importer,
+  Version,
+} from 'sass';
+
+import {captureStdio, URL} from './utils';
+
+/**
+ * Map from obsolete deprecation IDs to version pairs.
+ *
+ * The first version is the version this deprecation type was deprecated in,
+ * while the second version is the version it was made obsolete in.
+ */
+const obsoleteDeprecations: {[key: string]: [string, string]} = {};
+
+/** Map from active deprecation IDs to the version they were deprecated in. */
+const activeDeprecations = {
+  'call-string': '0.0.0',
+  elseif: '1.3.2',
+  'moz-document': '1.7.2',
+  'relative-canonical': '1.14.2',
+  'new-global': '1.17.2',
+  'color-module-compat': '1.23.0',
+  'slash-div': '1.33.0',
+  'bogus-combinators': '1.54.0',
+  'strict-unary': '1.55.0',
+  'function-units': '1.56.0',
+  'duplicate-var-flags': '1.62.0',
+  'null-alpha': '1.62.3',
+  'abs-percent': '1.65.0',
+  'fs-importer-cwd': '1.73.0',
+};
+
+/** List of future deprecation IDs. */
+const futureDeprecations = ['import'];
+
+/**
+ * This is a temporary synchronization check to ensure that any new deprecation
+ * types are added to all five of these locations:
+ * - lib/src/deprecation.dart in sass/dart-sass
+ * - js-api-doc/deprecations.d.ts in sass/sass
+ * - spec/js-api/deprecations.d.ts.md in sass/sass
+ * - lib/src/deprecations.ts in sass/embedded-host-node
+ * - js-api-spec/deprecations.test.ts in sass/sass-spec (this file)
+ *
+ * Work to replace these manual changes with generated code from a single
+ * source-of-truth is tracked in sass/sass#3827
+ */
+it('there are no extra deprecation types', () => {
+  const expectedDeprecations = [
+    ...Object.keys(obsoleteDeprecations),
+    ...Object.keys(activeDeprecations),
+    ...futureDeprecations,
+    'user-authored',
+  ];
+  const extraDeprecations = Object.keys(deprecations).filter(
+    deprecation => !expectedDeprecations.includes(deprecation)
+  );
+  expect(extraDeprecations).toBeEmptyArray();
```

Comment:
We should check that everything in `expectedDeprecations` is in `deprecations` as well. Also, we should _probably_ have some sort of test for the deprecation list in the JS API spec as well.

Reply:
Added an additional check that there are no missing deprecations either.

I also added some types which should ensure that there aren't any extra deprecations in these tests that aren't in the spec.